src/utils/channel_utils.js

```diff
@@ -390,13 +390,20 @@ export function sortChannelsByDisplayName(locale, a, b) {
         return (typeToPrefixMap[a.type] || defaultPrefix).localeCompare((typeToPrefixMap[b.type] || defaultPrefix), locale);
     }
 
-    if (a.display_name !== b.display_name) {
-        return a.display_name.toLowerCase().localeCompare(b.display_name.toLowerCase(), locale, {numeric: true});
+    const aDisplayName = filterName(a.display_name);
+    const bDisplayName = filterName(b.display_name);
+
+    if (aDisplayName !== bDisplayName) {
+        return aDisplayName.toLowerCase().localeCompare(bDisplayName.toLowerCase(), locale, {numeric: true});
     }
 
     return a.name.toLowerCase().localeCompare(b.name.toLowerCase(), locale, {numeric: true});
 }
 
+function filterName(name) {
+    return name.replace(/[.,'"\/#!$%\^&\*;:{}=\-_`~()]/g, ''); // eslint-disable-line no-useless-escape
```

Comment:
I hate that rule. Using a plain - inside one of those blocks is just asking for trouble if someone changes it and doesn't realize it